### PR TITLE
Implement typescript native driver

### PR DESCRIPTION
**Pull request by erizocosmico**
/cc @Serabe @smola @mcuadros 

P.D.: Can we pretty please give permissions in this org so we can assign reviewers instead of cc'ing?

### CODE REVIEW

**Comment by erizocosmico** [comment]
Updated with JSDoc comment parsing and non-JSDoc comment extraction.

**Review by smola — APPROVED** [review]

File: native/src/index.js [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+const { processRequest } = require('./parse');
+const readline = require('readline');
+
+const rl = readline.createInterface({
+  input: process.stdin,
+  output: process.stdout,
+  terminal: false
+});
+
+rl.on('line', l => console.log(processRequest(l)));
```
Comment (Serabe) [inline native/src/index.js:13]: I tend to prefer streams for this kind of transforming. No no-go. Just the solution for javascript will be different xD.

File: Dockerfile.build [hunk 2]
```diff
@@ -0,0 +1,27 @@
+# Dockerfile.build represents the build environment of the driver, used during
+# the development phase to test and in CI to build and test.
+
+# The prefered base image is the lastest stable Alpine image, if alpine doesn't
+# meet the requirements you can switch the from to the latest stable slim
+# version of Debian (eg.: `debian:jessie-slim`).
+FROM node:7-alpine
```
Comment (Serabe) [inline Dockerfile.build:7]: We need to talk about which version of node we should use. I was planning on using LTS, but open to other policies.

File: .travis.yml [hunk 3]
```diff
@@ -0,0 +1,18 @@
+language: go
```
Comment (Serabe) [inline .travis.yml:1]: I thought @mcuadros said we should not be using sdk yet.

File: README.md [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+Requirements:
+- `docker`
+- [`bblfsh-sdk`](https://github.com/bblfsh/sdk) _(go get -u github.com/bblfsh/sdk/...)_
+
+To initialize the build system execute: `bblfsh-sdk prepare-build`, at the root of the project. This will install the SDK at `.sdk` for this driver.
+
+To execute the tests just execute `make test`, this will execute the test over the native and the go components of the driver. Use `make test-native` to run the test only over the native component or `make test-driver` to run the test just over the go component.
+
+The build is done executing `make build`. To evaluate the result, a docker container, execute:
+`docker run -it bblfsh/java-driver:dev-<commit[:6]>`
```
Comment (Serabe) [inline README.md]: s/java/typescript/

This is solved in sdk now.

**Review by Serabe — CHANGES_REQUESTED** [review]

File: native/src/index.js (hunk 1 above)
Comment (erizocosmico) [reply native/src/index.js:13]: :trollface: now it's the same in javascript's, so I get i don't have to changes this

File: Dockerfile.build (hunk 2 above)
Comment (erizocosmico) [reply Dockerfile.build:7]: we can perfectly use the LTS, was the first that I found, tbh. Anyway, Dockerfiles are bound to change once the SDK is ready to use, so they will get changed and then we'll decide what version to use. But yeah, LTS makes sense

File: .travis.yml (hunk 3 above)
Comment (erizocosmico) [reply .travis.yml:1]: yep, but since it's already there /shrug

File: native/src/index.js (hunk 1 above)
Comment (Serabe) [reply native/src/index.js:13]: Yep, really all crappy comment of mine.

Why do all nasty advertisements launch a confirm dialog before leaving but Github does not?

**Review by Serabe — APPROVED** [review]

File: native/src/index.js (hunk 1 above)
Comment (erizocosmico) [reply native/src/index.js:13]: i guess we'll never know

**Pull request merged**